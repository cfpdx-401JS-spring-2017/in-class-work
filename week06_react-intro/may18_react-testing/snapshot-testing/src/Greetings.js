import React, { Component } from 'react';

export function Greeter({ name }) {
  return <div><span>Hello</span> <span>{name}</span></div>;
}

export class Greetings extends Component {
  constructor() {
    super();
    this.state = {
      people: ['Jane', 'Jill', 'Joey']
    };
  }

  render() {
    const { people } = this.state;
    return (
      <ul>
        {people.map((person, i) => <Greeter key={i} name={person} />)}
      </ul>
    );
  }
}
