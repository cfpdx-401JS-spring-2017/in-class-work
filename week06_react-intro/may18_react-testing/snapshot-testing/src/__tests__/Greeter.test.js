import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';

import { Greeter, Greetings } from '../Greetings';

describe('Greeter', function () {

    it('says hello', function () {
      const json = renderer.create(<Greeter name="Marty"/>).toJSON();
      expect(json).toMatchSnapshot();
    });
    
});

describe('Greetings', function () {

    it('greets all people', function () {
      const json = renderer.create(<Greetings/>).toJSON();
      expect(json).toMatchSnapshot();
    });
    
});