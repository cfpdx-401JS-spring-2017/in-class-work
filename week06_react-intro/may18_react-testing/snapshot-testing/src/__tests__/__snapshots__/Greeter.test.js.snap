exports[`Greeter says hello 1`] = `
<div>
  Hello 
  <span>
    Marty
  </span>
</div>
`;

exports[`Greetings greets all people 1`] = `
<ul>
  <Greeter
    name="Jane" />
  <Greeter
    name="Jill" />
  <Greeter
    name="Joey" />
</ul>
`;
